internal/provider/data_source_certificate.go

```diff
@@ -115,51 +124,69 @@ func dataSourceCertificate() *schema.Resource {
 func dataSourceCertificateRead(d *schema.ResourceData, m interface{}) error {
 	config := m.(*providerConfig)
 
-	targetURL, err := url.Parse(d.Get("url").(string))
-	if err != nil {
-		return err
-	}
-
-	// Determine if we should verify the chain of certificates, or skip said verification
-	shouldVerifyChain := d.Get("verify_chain").(bool)
-
-	// Ensure a port is set on the URL, or return an error
-	var peerCerts []*x509.Certificate
-	switch targetURL.Scheme {
-	case HTTPSScheme.String():
-		if targetURL.Port() == "" {
-			targetURL.Host += ":443"
+	if v, ok := d.GetOk("content"); ok {
+		block, _ := pem.Decode([]byte(v.(string)))
+		if block == nil {
+			return fmt.Errorf("failed to decode pem content")
+		}
+		if block.Type != "CERTIFICATE" {
```

Comment:
Please use the `PEMPreamble` const available in `types.go`.

Reply:
You should also use `PEMBlockToPEMPreamble` to extract the correct `PEMPreamble`.